src/tests/multihost/ipa/test_subid_ranges.py

```diff
@@ -0,0 +1,132 @@
+""" Automation of IPA bugs """
+
+import pytest
+import subprocess
+import time
+import os
+from sssd.testlib.common.utils import SSHClient
+
+
+test_password = "Secret123"
+user = 'admin'
+
+
+def execute_cmd(multihost, command):
+    """ Execute command on client """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.usefixtures('install_list_subid_ranges',
+                         'setup_ipa_client')
+@pytest.mark.tier1
+class TestSubid(object):
+    """
+    This is for ipa bugs automation
+    """
+    def test_subid_feature(self, multihost,
+                           create_bkp_for_subid_files):
+        """
+        :Title: support subid ranges managed by FreeIPA
+        :id: 50bcdc28-00c8-11ec-bef4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1803943
+        :steps:
+            1. Generate subid for user admin
+            2. Test newuidmap command
+            3. Test newgidmap command
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        (result, result1, exit_status) = ssh1.execute_cmd('kinit',
+                                                          stdin=test_password)
+        assert exit_status == 0
+        (result, result1, exit_status) = ssh1.exec_command('klist')
+        assert user in str(result1.read())
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                           f"subid-generate "
+                                                           f"--owner={user}")
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                           f"subid-find "
+                                                           f"--owner "
+                                                           f"{user}")
+        user_details = result1.readlines()
+        uid_start = int(user_details[5].split(': ')[1].split('\n')[0])
+        uid_range = int(user_details[6].split(': ')[1].split('\n')[0])
+        gid_start = int(user_details[7].split(': ')[1].split('\n')[0])
+        gid_range = int(user_details[8].split(': ')[1].split('\n')[0])
+        ssh2 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        cmd = execute_cmd(multihost, "ps -ef | grep bash").stdout_text
+        find_admin = f"ps -ef | grep bash | grep {user}"
+        if user in cmd:
+            proces_id = [int(i)
+                         for i in execute_cmd(multihost,
+                                              find_admin).stdout_text.split()
+                         if i.isdigit()][0]
+            multihost.client[0].run_command(f'kill -9 {proces_id}')
```

Comment:
What is the source of those shells (bash) that needs to be terminated?

Reply:
Here i am making sure that there is no previous process id exits for user admin

Reply:
That's clear.
The questions are:
 - what is the source of those processes? How did they appear?
 - Why is it important to terminate those? It would be enough to figure out `pid` of proper shell later.

Reply:
There might be some other process running for admin from previous test , we dont know when this test script will be executed and after which test . So for this test to work correctly , here i am making sure that no other admin process are there by killing any other admin user process

Reply:
something like `unshare -U bash -c 'echo $$ > /tmp/unshare.pid; sleep 10'`
just replace sleep 10 with something like `while(true) sleep(1)`

Reply:
fixed